fil-proofs-tooling/src/bin/benchy/zigzag.rs

```diff
@@ -89,6 +90,27 @@ impl From<Params> for Inputs {
     }
 }
 
+struct FuncMeasurement<T> {
+    cpu_time: Duration,
+    wall_time: Duration,
+    return_value: T,
+}
+
+fn measure<T, F: FnOnce() -> Result<T, failure::Error>>(
+    f: F,
+) -> Result<FuncMeasurement<T>, failure::Error> {
+    let cpu_time_start = ProcessTime::now();
+    let wall_start_time = Instant::now();
+
+    let x = f()?;
+
+    Ok(FuncMeasurement {
+        cpu_time: cpu_time_start.elapsed(),
+        wall_time: wall_start_time.elapsed(),
+        return_value: x,
+    })
+}
```

Comment:
I think a nicer api, which avoids wrapping things in a closure would be this

```rust
struct RunningMeasurement {
  cpu_time: ProcessTime,
  wall_time: Instant,
}

fn start() -> RunningMeasurement {
  RunningMeasurement {
    cpu_time: ProcessTime::now(),
    wall_start_time: Instant::now(),
  }
}

struct RunningMeasurement {
  cpu_time: Duration,
  wall_time: Duration
}

impl RunningMeasurement {
  fn stop(self) -> Measurement {
    Measurement {
      cpu_time: self.cpu_time.elapsed(),
	  wall_time: self.wall_time.elapsed()
    }
  }
}

// usage
let m1 = start();
do_expensive_work();
let result = m1.stop();
```

Reply:
also, either in this PR or in the next, lets start putting these helpers into a shared module, so we can reuse them between different benchmarks

Reply:
> which avoids wrapping things in a closure would be this

I wanted to wrap it in a closure. I don't want to have to deal with a bunch of stop/start calls.